pickle/src/com/treode/pickle/PicklerRegistry.scala

```diff
@@ -14,100 +14,269 @@
  * limitations under the License.
  */
 
-package com.treode.pickle
+ package com.treode.pickle
 
-import java.util.concurrent.ConcurrentHashMap
-import scala.reflect.ClassTag
+ import java.util.concurrent.ConcurrentHashMap
+ import scala.reflect.ClassTag
 
-import com.treode.buffer.{InputBuffer, PagedBuffer}
+ import com.treode.buffer.{InputBuffer, PagedBuffer}
 
-import PicklerRegistry._
+ import PicklerRegistry._
 
-class PicklerRegistry [T] private (default: Long => T) {
+ class PicklerRegistry [T] private (default: Long => T) {
 
   private val openers = new ConcurrentHashMap [Long, Opener [T]]
 
-  def register [P] (p: Pickler [P], tag: Long) (read: P => T) {
-    val u1 = Opener (p, tag, read)
-    val u0 = openers.putIfAbsent (tag, u1)
-    require (u0 == null, f"$tag%X already registered")
+  /**
+ *  The register method registers a new Pickler with the id ``tag``.
```

Comment:
I am particular about documentation (when it's present). 

"The register method..." is redundant; of course this about the register method. 

Hit the highlights in the first sentence, because Scaladoc treats that as a one line summary; it shows just the first sentence until the user expands the doc for that method, so the first sentence needs to get straight to the point.

Only single backticks are needed, so it's `tag`.